lab-yunjoo/server.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const http = require('http');
+const cowsay = require('cowsay');
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const parseBody = require('./lib/parse-json');
+const port = process.argv[2]||3000;
+
+var server = http.createServer(function(req,res){
+  req.url = parseUrl(req.url);
+  var pathname = req.url.pathname;
+  var query = req.url.query = parseQuery(req.url.query);
+  if(pathname == '/'){
+    res.writeHead(200,{'Content-Type' : 'text/plain'});
+    res.write(cowsay.say({text:'Welcome to The cow world'}));
+    res.end();
+    return;
+  } else if(pathname == '/cowsay') {
+    if(req.method == 'GET'){
```

Comment:
According to the rubric, there needs to be a section that throws a 400 error if the query string is undefined.

"if the query text=message is not set, respond with:
status code = 400
a body including the value returned from cowsay.say({text: 'bad request\ntry: localhost:3000/cowsay?text=howdy'})
"